pkg/applicationserver/applicationserver.go

```diff
@@ -786,8 +791,13 @@ var errFetchAppSKey = errors.Define("app_s_key", "failed to get AppSKey")
 // handleJoinAccept handles a join-accept message.
 // If the application or device is not configured to skip application crypto, the InvalidatedDownlinks and the AppSKey
 // in the given join-accept message is reset.
-func (as *ApplicationServer) handleJoinAccept(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers, joinAccept *ttnpb.ApplicationJoinAccept, link *ttnpb.ApplicationLink) error {
+func (as *ApplicationServer) handleJoinAccept(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers, joinAccept *ttnpb.ApplicationJoinAccept, link *ttnpb.ApplicationLink) (err error) {
 	defer trace.StartRegion(ctx, "handle join accept").End()
+	defer func() {
+		if err == nil {
+			go as.lastSeenMap.storeLastSeenFromUplink(ctx, ids, joinAccept.ReceivedAt)
```

Comment:
Please use a `workerpool.Workerpool` - look at how the device activations are currently sent to the IS